### PR TITLE
[firebase-analytics] update doc comments, extract API data

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1904

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `FirebaseAnalytics` API - https://docs.expo.io/versions/latest/sdk/firebase-analytics - with minor tweaks and additions to the package source code.

To full support the migration to automated docs extraction I had to introduce few tweaks to the code:
* generic object types has been replaced with `Record`s
* doc parameter names has been corrected, to match updated naming

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Preview

Screenshot 2021-09-06 at 12 17 47

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-firebase-analytics/src/Analytics.ts [hunk 1]
```diff
@@ -196,12 +222,11 @@ export function setClientId(clientId: string): void {
 }
 
 /**
- * Enables or disabled debug mode on the Expo client, so events can
+ * __Expo Go Only.__ Enables or disabled debug mode on the Expo client, so events can
```
Comment (EvanBacon) [inline packages/expo-firebase-analytics/src/Analytics.ts:225]: Wonder if maybe we should add a thing like `@runtime` that we can use with `Expo Go/Custom` or maybe some other indication like the Expo Go version.

File: packages/expo-firebase-analytics/src/Analytics.ts [hunk 2]
```diff
@@ -196,12 +222,11 @@ export function setClientId(clientId: string): void {
...
  * be tracked using the [DebugView in the Analytics dashboard](https://firebase.google.com/docs/analytics/debugview#reporting).
  *
- * This option is only available on the standard Expo client.
- * When using a standalone build, the bare workflow or web, use the
- * [natively available options](https://firebase.google.com/docs/analytics/debugview).
+ * This option is only available on the standard Expo client. When using a standalone build, the
```
Comment (ide) [inline packages/expo-firebase-analytics/src/Analytics.ts:228]:
```suggestion
 * This option is only available in Expo Go. When using a custom development app, a standalone app, the
```

File: packages/expo-firebase-analytics/src/Analytics.ts (hunk 1 above)
Comment (Simek) [reply packages/expo-firebase-analytics/src/Analytics.ts:225]: That's a good idea, I have the platform specific annotations (`@ios`, `@android`, `@web`) on the TODO list already, but I wanted to wait for the docs redesign, which will rewrite a lot of UI code and should include a component which can be easily used as label in docs. 

So currently I just stick to plain text information, which can been seen in the docs before I have started the work on `APISection`. 

In this specific case the docs have this annotation already, but the code comment did not:
* https://github.com/expo/expo/pull/14302/files#diff-2f1f696c927c4e572b9cf5a22dd9ebb3b7bfe677c2cf10ffcaa0bbed7be78eb5L326

**Review by IjzerenHein — APPROVED** [review]
Apart from the other comments, look good to me 👍

**Review by bbarthec — APPROVED** [review]
LGTM 👍

**Pull request merged**